js/modules/k6/http/http.go

```diff
@@ -21,101 +21,152 @@
 package http
 
 import (
-	"context"
+	"net/http"
+	"net/http/cookiejar"
 
+	"github.com/dop251/goja"
 	"go.k6.io/k6/js/common"
-	"go.k6.io/k6/lib"
+	"go.k6.io/k6/js/modules"
 	"go.k6.io/k6/lib/netext"
 	"go.k6.io/k6/lib/netext/httpext"
 )
 
-const (
-	HTTP_METHOD_GET     = "GET"
-	HTTP_METHOD_POST    = "POST"
-	HTTP_METHOD_PUT     = "PUT"
-	HTTP_METHOD_DELETE  = "DELETE"
-	HTTP_METHOD_HEAD    = "HEAD"
-	HTTP_METHOD_PATCH   = "PATCH"
-	HTTP_METHOD_OPTIONS = "OPTIONS"
-)
+// RootModule is the global module object type. It is instantiated once per test
+// run and will be used to create HTTP module instances for each VU.
+//
+// TODO: add sync.Once for all of the deprecation warnings we might want to do
+// for the old k6/http APIs here, so they are shown only once in a test run.
+type RootModule struct{}
+
+// ModuleInstance represents an instance of the HTTP module for every VU.
+type ModuleInstance struct {
+	vu            modules.VU
+	rootModule    *RootModule
+	defaultClient *Client
```

Comment:
```suggestion
	client *Client
```
Do we expect to have a dedicated field for non-default? Otherwise I think we can remove it.

Reply:
Yes, we expect to be able to create non-default HTTP clients with different options and transports in the near (:crossed_fingers:) future. So you'd be able do to something like this:
```js
import http from 'k6/http';
let myAwesomeTransport = new http.Transport({forceHTTP1: true}) // or something like this
let myAwesomeClient = new http.Client(myAwesomeTransport);
```

Reply:
What I meant is that from my understanding `ModuleInstance` will have only a unique field as `client` so we could avoid to explicit `default` because that should be clear and it will be wrapped behind the exported methods `get/post/head etc...`. Btw it's just a preference, not really required.